src/IRrecv.cpp

```diff
@@ -132,21 +132,39 @@ void IRrecv::resume() {
   irparams.overflow = false;
 }
 
-// Make a copy of the interrupt state/data.
+// Make a copy of the interrupt state & buffer data.
 // Needed because irparams is marked as volatile, thus memcpy() isn't allowed.
 // Only call this when you know the interrupt handlers won't modify anything.
 // i.e. In STATE_STOP.
 //
 // Args:
-//   dest: Pointer to an irparams_t structure to copy to.
-void IRrecv::copyIrParams(irparams_t *dest) {
-  // Typecast src and dest addresses to (char *)
-  char *csrc = (char *) (&irparams);  // NOLINT(readability/casting)
-  char *cdest = (char *) dest;  // NOLINT(readability/casting)
-
-  // Copy contents of src[] to dest[]
+//   src: Pointer to an irparams_t structure to copy from.
+//   dst: Pointer to an irparams_t structure to copy to.
+void IRrecv::copyIrParams(volatile irparams_t *src, irparams_t *dst) {
+  // Typecast src and dst addresses to (char *)
+  char *csrc = (char *) src;  // NOLINT(readability/casting)
+  char *cdst = (char *) dst;  // NOLINT(readability/casting)
+
+  // Save the pointer to the destination's rawbuf so we don't lose it.
+  uint16_t *dst_rawbuf_ptr;
+  dst_rawbuf_ptr = dst->rawbuf;
+
+  // Copy contents of src[] to dst[]
   for (uint16_t i = 0; i < sizeof(irparams_t); i++)
-    cdest[i] = csrc[i];
+    cdst[i] = csrc[i];
+
+  // Restore the buffer pointer
+  dst->rawbuf = dst_rawbuf_ptr;
```

Comment:
Looks can be deceiving. :) Lines 153-154 copy right over the top of it.
I'll add a comment to make that more clear.